crates/roc_std/src/roc_str.rs

```diff
@@ -380,9 +382,7 @@ impl RocStr {
         self.with_terminator(terminator, |dest_ptr: *mut u16, str_slice: &str| {
             // Translate UTF-8 source bytes into UTF-16 and write them into the destination.
             for (index, wchar) in str_slice.encode_utf16().enumerate() {
-                unsafe {
-                    *(dest_ptr.add(index)) = wchar;
-                }
+                unsafe { std::ptr::write_unaligned(dest_ptr.add(index), wchar) };
```

Comment:
the alignment of the input pointer is often incorrect in our tests. Miri finds this. Recent versions of the rust compiler will likely also find this and panic. The proper fix might be to make the input aligned, but that is non-trivial

Reply:
Is it only in our tests? If so, can we do `#[cfg(test)]` for this? (Just because there's a runtime performance penalty for unaligned writes.)

Reply:
I believe the penalty is not really there on modern hardware if the write does end up being aligned